pkg/ast/spl/tests/splParser_test.go

```diff
@@ -262,6 +262,40 @@ func Test_searchFieldGreaterThanOrEqualToQuotedString(t *testing.T) {
 	assert.NotNil(t, err)
 }
 
+func Test_searchFieldEqualToBooleanTrue(t *testing.T) {
+	query := []byte(`search status=true`)
+	res, err := spl.Parse("", query)
+	assert.Nil(t, err)
+	filterNode := res.(ast.QueryStruct).SearchFilter
+
+	assert.NotNil(t, filterNode)
+	assert.Equal(t, "=", filterNode.Comparison.Op)
+	assert.Equal(t, "status", filterNode.Comparison.Field)
+	assert.Equal(t, true, filterNode.Comparison.Values)
+
+	expressionFilter := extractExpressionFilter(t, filterNode)
+	assert.Equal(t, "status", expressionFilter.LeftInput.Expression.LeftInput.ColumnName)
+	assert.Equal(t, utils.Equals, expressionFilter.FilterOperator)
+	assert.Equal(t, "true", expressionFilter.RightInput.Expression.LeftInput.ColumnValue.StringVal)
```

Comment:
The value should be a boolean, so you should probably assert the `ColumnValue.BoolVal` instead of the `ColumnValue.StringVal`. You can also check that the `ColumnValue.Dtype` is `SS_DT_BOOL`

Same comment for line 296

Reply:
@AndrewHess i made the changes